### PR TITLE
Feature: Soft locking

### PR DESCRIPTION
**Pull request by mindofmar**
image height=300px

Introducing soft locking: keep your most valuable assets locked down and safe from scammers.

### CODE REVIEW

File: src/models/simulation/ChangeTypeHelper.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { StateChange, SimulationChangeType, AssetKey } from './Transaction';
+
+export function AssetsEqual(assetKey: AssetKey, stateChange: StateChange): boolean {
+  return assetKey.contractAddress === stateChange.contractAddress &&
+    assetKey.ercType === stateChange.assetType &&
+    (assetKey.tokenId === stateChange.tokenId || assetKey.tokenId === 'COLLECTION');
+}
+
+export function IsTransferChangeType(changeType: SimulationChangeType): boolean {
```
Comment (AugmentedMode) [inline src/models/simulation/ChangeTypeHelper.ts:17]: love this

File: src/lib/simulation/storage.ts [hunk 2]
```diff
@@ -181,6 +221,7 @@ export const fetchSimulationAndUpdate = async (args: TransactionArgs) => {
         args,
       }),
       fetchTransaction(args, TransactionType.Signature),
+      // do not fetch locked assets on personal_sign
```
Comment (AugmentedMode) [inline src/lib/simulation/storage.ts:224]: so we fetch locked assets on signatures and transaction. just not personal sign right?

File: src/components/simulation/SimulationSubComponents/errors/LockedAssetPopup.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  }, []);
+
+  return (
+    <div className='container' style={{ position: 'fixed', bottom: '100px' }}>
+      <div style={{ width: '100%', borderRadius: '10px', display: 'flex', flexDirection: 'row', alignItems: 'center', background: '#2B2B2B', padding: '15px' }}>
+        <img src='/images/popup/notAllowedIcon.png' alt='a not allowed icon' />
+
+        <div style={{ display: 'flex', flexDirection: 'column', marginLeft: '20px' }}>
+          <p style={{ fontSize: '20px', marginBottom: '0px' }} className={styles['font-archivo-semibold']}>Locked asset detected</p>
+          <p className={styles['softLockedWarningSubtitle']} style={{ marginBottom: '0px' }}>If this was intentional, visit your dashboard to unlock the asset.</p>
```
Comment (AugmentedMode) [inline src/components/simulation/SimulationSubComponents/errors/LockedAssetPopup.tsx:17]: this popup contains a go to dashboard button right?

File: src/components/simulation/SimulationSubComponents/StateChangesComponent.tsx
Comment (AugmentedMode) [inline src/components/simulation/SimulationSubComponents/StateChangesComponent.tsx:1]: which ones are locked asset not apart of?

**Review by AugmentedMode — DISMISSED** [review]

File: src/lib/simulation/storage.ts (hunk 2 above)
Comment (mindofmar) [reply src/lib/simulation/storage.ts:224]: correct

File: src/components/simulation/SimulationSubComponents/errors/LockedAssetPopup.tsx (hunk 3 above)
Comment (mindofmar) [reply src/components/simulation/SimulationSubComponents/errors/LockedAssetPopup.tsx:17]: No but the continue button is replaced with Go to Dashboard

**Review by AugmentedMode — DISMISSED** [review]

**Review by AugmentedMode — APPROVED** [review]

**Pull request merged**